packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -51,13 +56,21 @@ String getDefaultCachedKernelPath({
   return getKernelPathForTransformerOptions(
     globals.fs.path.join(getBuildDirectory(), '${buildPrefix}cache.dill'),
     trackWidgetCreation: trackWidgetCreation,
+    nullSafetyMode: nullSafetyMode,
   );
 }
 
 String getKernelPathForTransformerOptions(
   String path, {
   @required bool trackWidgetCreation,
+  @required NullSafetyMode nullSafetyMode,
 }) {
+  // Temporary work around until --initialize-from-dill accounts for sound mode.
```

Comment:
Could you file an issue for this?

Reply:
Filled https://github.com/flutter/flutter/issues/68901